src/otg/dataset/study_locus.py

```diff
@@ -368,6 +369,22 @@ def annotate_credible_sets(self: StudyLocus) -> StudyLocus:
         )
         return self
 
+    def annotate_ld(
+        self: StudyLocus, study_index: StudyIndex, ld_index: LDIndex
+    ) -> StudyLocus:
+        """Annotate LD information to study-locus.
+
+        Args:
+            study_index (StudyIndex): Study index to resolve ancestries.
+            ld_index (LDIndex): LD index to resolve LD information.
+
+        Returns:
+            StudyLocus: Study locus annotated with ld information from LD index.
+        """
+        from otg.method.ld import LDAnnotator
```

Comment:
Having an import statement in a function definition seems quite uncanonical to me. Is it intentional?

Reply:
It's a common practice to avoid the problem of dependency circularity

LD annotator imports from StudyLocus
StudyLocus imports from LD annotator 

Importing LDAnnotator only when it's needed removes this. You're right in that in general it's more pythonic having them at the top.